core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -463,21 +499,28 @@ public boolean ownsData() {
                }
             }
 
-            try {
-               removeStaleData(removedSegments);
-            } catch (InterruptedException e) {
-               Thread.currentThread().interrupt();
-               throw new CacheException(e);
-            }
-            conflictManager.restartVersionRequests();
+            return removeStaleData(removedSegments)
+                  .thenCompose(ignored1 -> {
+                     conflictManager.restartVersionRequests();
+
+                     // preserve the
```

Comment:
Yeah, we'd lose the throwable, but I don't think it was any different in the synchronous version, where an exception in `finally` would hide the original exception.

I do want to change this area so it's more like a state machine instead of a chain of lambdas, but there was no chance for that to make it into 10.1 :)